src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -168,6 +168,33 @@ public void validate(ClientState state)
             statement.validate(state);
     }
 
+    @Override
+    public Iterable<CQLStatement> getStatements()
+    {
+        return () -> {
+            Stream<CQLStatement> stream = assignments.stream().map(n -> n.select);
+            if (returningSelect != null)
+                stream = Stream.concat(stream, Stream.of(returningSelect.select));
+            stream = Stream.concat(stream, updates.stream());
+            return stream.iterator();
+        };
+    }
+
+    @Override
+    public ResultSet.ResultMetadata getResultMetadata()
+    {
+        if (returningSelect != null)
```

Comment:
Result metadata seems to be something final and perhaps it should be created in the constructor, wdyt?

Reply:
true we could, but I don't think we need to as the access is short lived here.  prepared statements holds ref in `org.apache.cassandra.cql3.QueryHandler.Prepared#Prepared` and txn statement calls this after we get the response back from the accord coordinator, so by keeping it something we "compute" we keep this object in short term memory rather than make it used 2 times in long term memory

Reply:
to clarify, in the prepared case we would compute this once and reuse in the `execute` response, but when not prepared you compute this on-demand and this is short lived...

thinking about this more maybe we should, so in the prepared case we only call `org.apache.cassandra.cql3.ResultSet#computeResultMetadataId` once and the non-prepared case we also only call once...  right now prepared case will call this the same amount as non-prepared and an additional one while prepare...  given that `org.apache.cassandra.cql3.ResultSet#computeResultMetadataId` does have a noticeable impact on perf, doing once would help the prepared case and be equal to the non-prepared case... ill make this change

Reply:
made this change, we now compute this once per object, so prepared statements should not be hashing the names